src/components/EditActivityPage/EditActivityPage.js

```diff
@@ -50,12 +53,22 @@ const EditActivityPage = ({ activity, history, onStartEditActivity, onStartRemov
 };
 
 EditActivityPage.propTypes = {
-  activity: PropTypes.shape(activityPropTypeShape).isRequired,
+  activity: PropTypes.shape(activityPropTypeShape),
   history: PropTypes.shape(historyPushPropTypeShape).isRequired,
   onStartEditActivity: PropTypes.func.isRequired,
   onStartRemoveActivity: PropTypes.func.isRequired,
 };
 
+EditActivityPage.defaultProps = {
+  activity: {
```

Comment:
if there can be no activity, there shouldn't be such a fake object. it would be better to make `activity: {}`. 

If you have some problems due to that fact, you should guard activity and it's props. `get` from `lodash` can help you.